lib/src/main/java/com/diffplug/spotless/generic/NativeCmdStep.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.generic;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import com.diffplug.spotless.FileSignature;
+import com.diffplug.spotless.FormatterFunc;
+import com.diffplug.spotless.FormatterStep;
+import com.diffplug.spotless.ProcessRunner;
+
+public class NativeCmdStep {
+	// prevent direct instantiation
+	private NativeCmdStep() {}
+
+	public static FormatterStep create(String name, File pathToExe, List<String> arguments) {
+		Objects.requireNonNull(name, "name");
+		Objects.requireNonNull(pathToExe, "pathToExe");
+		List<String> argumentsWithPathToExe = new ArrayList<>();
+		argumentsWithPathToExe.add(pathToExe.getAbsolutePath());
```

Comment:
By adding the absolute path here, you're making `State` machine-dependent, but it doesn't have to be. You can use `FileSignature.getOnlyFile()` to build the full `List<String>` within `toFunc`. Then the step will be compatible with remote build cache.

Reply:
I'm about to cut a release, if you don't care about remote build-cache, I'm okay to merge this one as-is. You've submitted three great PRs and they've gotten bogged down in our CI hiccup, and this is already a good PR as-is.

Reply:
Give me a minute 😄

Reply:
Changed it, so that the absolute path is evaluated later